src/shared/test/integration/rpc-bridge-test.js

```diff
@@ -0,0 +1,206 @@
+      const bridge = createBridge();
+      const reciprocalBridge = createBridge();
+      const errorMessage = 'My error';
+      bridge.on('method1', (_arg, cb) => cb(errorMessage));
+      bridge.createChannel(port1);
+      const channel = reciprocalBridge.createChannel(port2);
+      sandbox.stub(channel, 'destroy').callThrough();
+
+      let error;
+      try {
+        await reciprocalBridge.call('method1', 'params1');
+      } catch (err) {
+        error = err;
+      }
+
+      assert.called(channel.destroy);
+      assert.equal(error, errorMessage);
+      assert.deepEqual(reciprocalBridge.links, []);
+    });
+
+    it('no longer send RPC messages to a Bridge channel that has received an error', async () => {
+      const bridge = createBridge();
+      const reciprocalBridge = createBridge();
+      const errorMessage = 'My error';
+      bridge.on('method1', (_arg, cb) => cb(new Error(errorMessage)));
+      bridge.createChannel(port1);
+      const channel = reciprocalBridge.createChannel(port2);
+      sandbox.stub(channel, 'call').callThrough();
+
+      let error;
+      try {
+        await reciprocalBridge.call('method1', 'params1');
+      } catch (err) {
+        error = err;
+      }
+      assert.equal(error.message, errorMessage);
+
+      for (let i = 0; i < 5; ++i) {
+        const results = await reciprocalBridge.call('method1', 'params1');
+        assert.deepEqual(results, []);
+        assert.calledOnce(channel.call);
+      }
+    });
+
+    it('timeouts if the Bridge channel is not connected', done => {
+      const bridge = createBridge();
+      bridge.createChannel(port1);
+
+      bridge.call('method1', 'params1', (err, results) => {
+        assert.deepEqual(results, [null]); // returns null for each channel that timeouts
+        assert.isNull(err); // no error
+        done();
+      });
+      clock.tick(1000);
+    });
+
+    it(`timeouts if the Bridge channel is connected but reciprocal Bridge channel doesn't answer`, done => {
+      const bridge = createBridge();
+      const reciprocalBridge = createBridge();
+      bridge.createChannel(port1);
+      reciprocalBridge.createChannel(port2); // the reciprocal port hasn't registered a RPC method called 'method1'
+
+      bridge.call('method1', 'params1', (err, results) => {
+        assert.deepEqual(results, [null]);
+        assert.isNull(err);
+        done();
+      });
+      clock.tick(1000);
+    });
+
+    it('timeouts if the reciprocal Bridge channel has been destroyed', done => {
+      const bridge = createBridge();
+      const reciprocalBridge = createBridge();
+      bridge.on('method1', (arg, cb) => cb(null, `${arg}foo`));
+      reciprocalBridge.on('method2', (arg, cb) => cb(null, `${arg}bar`));
+      bridge.createChannel(port1);
+      reciprocalBridge.createChannel(port2);
+
+      reciprocalBridge.destroy();
+
+      bridge.call('method2', 'params1', (err, results) => {
```

Comment:
As with the previous test, if we can use the Promise-based API here this will make it easier to remove the callback API in future.